File: rosecsharp/Subscription.cs
using System;

namespace Rose
{
    public class Subscription
    {
        private readonly Action unsubscribeAction;

        public Subscription(Action unsubscribeAction)
        {
            this.unsubscribeAction = unsubscribeAction;
        }

        public void Unsubscribe()
        {
            unsubscribeAction.Invoke();
        }
    }
}

File: java/com/kkucharski/rose/Subscription.java
package com.kkucharski.rose;

public class Subscription {
    private final Action unsubscribeAction;

    public Subscription(Action unsubscribeAction)
    {
        this.unsubscribeAction = unsubscribeAction;
    }

    public void unsubscribe()
    {
        unsubscribeAction.call();
    }
}
